pandas/tests/indexes/test_multi.py

```diff
@@ -963,19 +963,21 @@ def test_get_level_values(self):
         exp = CategoricalIndex([1, 2, 3, 1, 2, 3])
         tm.assert_index_equal(index.get_level_values(1), exp)
 
-    def test_get_level_values_na(self):
+    @pytest.mark.xfail(reason='GH 17924 (returns Int64Index with float data)')
```

Comment:
Why do we xfail this? (I mean, why don't we keep asserting for now that it is float)

Reply:
The assertions compare a correctly built index (``float64`` data in ``Float64Index``) with an invalid index (``float64`` data in ``Int64Index``, due to #17924 ). Maybe I'm missing something, but it's not obvious to me what we would assert - plus, I see it as an added value to ``xfail`` for a buggy behavior.

Reply:
OK